File: app/src/main/java/com/app/syspoint/repository/request/http/ApiServices.java
package com.app.syspoint.repository.request.http;

import static com.app.syspoint.utils.Constants.BASE_URL_DONAQUI;
import static com.app.syspoint.utils.Constants.BASE_URL_SYSPOINT;

import com.app.syspoint.BuildConfig;
import com.app.syspoint.utils.Constants;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiServices {

    private static Retrofit retrofit = null;

    public static Retrofit getClientRestrofit(){

        OkHttpClient okHttpClient = new OkHttpClient.Builder()
                .connectTimeout(1, TimeUnit.MINUTES)
                .readTimeout(30, TimeUnit.SECONDS)
                .writeTimeout(15, TimeUnit.SECONDS)
                .build();

        if (retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(getBaseURL())
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

    private static String getBaseURL() {
        if (BuildConfig.FLAVOR.equals("donaqui")) {
            return BASE_URL_DONAQUI;
        } else {
            return BASE_URL_SYSPOINT;
        }
    }

}

File: app/src/main/java/com/app/syspoint/repository/request/http/ApiServices.kt
package com.app.syspoint.repository.request.http

import com.app.syspoint.BuildConfig
import com.app.syspoint.utils.Constants
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import tech.thdev.network.flowcalladapterfactory.FlowCallAdapterFactory
import java.util.concurrent.TimeUnit

class ApiServices {
    companion object {
        private var retrofit: Retrofit? = null

        fun getClientRetrofit(): Retrofit {
            val okHttpClient = OkHttpClient.Builder()
                .connectTimeout(1, TimeUnit.MINUTES)
                .readTimeout(10, TimeUnit.SECONDS)
                .writeTimeout(10, TimeUnit.SECONDS)
                .build()

            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(getBaseURL())
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(FlowCallAdapterFactory())
                    .build()
            }
            return retrofit!!
        }

        private fun getBaseURL(): String {
            return Constants.BASE_URL_TENET_PROD
        }
    }
}